### PR TITLE
Bugfix/large sim jinja exception

### PR DESCRIPTION
**Pull request by nagkumar91**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/synthetic/_conversation/__init__.py [hunk 1]
```diff
@@ -85,9 +85,12 @@ def __init__(
                 if isinstance(conversation_starter_content, dict):
                     self.conversation_starter = conversation_starter_content
                 else:
-                    self.conversation_starter = jinja2.Template(
-                        conversation_starter_content, undefined=jinja2.StrictUndefined
-                    )
+                    try:
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/synthetic/_conversation/__init__.py:88]: do we need a unit test for covering this case?

**Review by ninghu — APPROVED** [review]

File: src/promptflow-evals/promptflow/evals/synthetic/_conversation/__init__.py (hunk 1 above)
Comment (nagkumar91) [reply src/promptflow-evals/promptflow/evals/synthetic/_conversation/__init__.py:88]: Just did! Thanks :)

**Review by ninghu — APPROVED** [review]

File: src/promptflow-evals/promptflow/evals/synthetic/_conversation/__init__.py [hunk 2]
```diff
@@ -85,9 +85,12 @@ def __init__(
...
                     self.conversation_starter = conversation_starter_content
                 else:
-                    self.conversation_starter = jinja2.Template(
-                        conversation_starter_content, undefined=jinja2.StrictUndefined
-                    )
+                    try:
+                        self.conversation_starter = jinja2.Template(
+                            conversation_starter_content, undefined=jinja2.StrictUndefined
+                        )
+                    except jinja2.exceptions.TemplateSyntaxError as e:  # noqa: F841
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/synthetic/_conversation/__init__.py:92]: If this is due to user error then do we just use the template with error ?

Comment (nagkumar91) [reply src/promptflow-evals/promptflow/evals/synthetic/_conversation/__init__.py:92]: `conversation_starter_content` is coming from the RAI service

Comment (singankit) [reply src/promptflow-evals/promptflow/evals/synthetic/_conversation/__init__.py:92]: Is it expected that template can be syntactically incorrect ?

**Review by singankit — APPROVED** [review]

**Pull request merged**